packages/@react-native-windows/cli/src/generator-windows/index.ts

```diff
@@ -95,6 +95,34 @@ export async function copyProjectTemplateAndReplace(
     namespace = namespace.split(/[.:]+/).map(pascalCase).join('.');
   }
 
+  // Checking if we're overwriting an existing project and re-uses their projectGUID
```

Comment:
This function is already really hairy. Could we pull this logic out into something separate?

Reply:
Per Jon's comments, this logic has been simplified to a single if-statment, let me know if you still think it should be pulled out into another function! :)